### PR TITLE
Add event memory

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Method `on` now support event memory

### CODE REVIEW

File: projects/engine/src/engine/engine.ts [hunk 1]
```diff
@@ -57,6 +58,10 @@ export class Engine {
       }
       this.events[listener][eventName](...payload)
     })
+    // Add even memory
```
Comment (yann300) [inline projects/engine/src/engine/engine.ts:61]: typo

File: projects/engine/src/engine/engine.ts [hunk 2]
```diff
@@ -57,6 +58,10 @@ export class Engine {
...
+    this.eventMemory[emitter]
+      ? this.eventMemory[emitter][event] = payload
+      : this.eventMemory[emitter] = { [event]: payload }
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply projects/engine/src/engine/engine.ts:64]: Here `[event]` is the key in the object `{ [event]: payload }`. It's like `this.eventMemory[emitter][event]` but in this case `this.eventMemory[emitter]` didn't already exist

[the remark this answers is not in the document]
Comment (yann300) [reply projects/engine/src/engine/engine.ts:64]: yes just that `{ [event]: payload }` looked like you are pushing an array which has `event` as first item
although you may want to have something like `{ event: payload }`

[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply projects/engine/src/engine/engine.ts:64]: If I do `{ event: payload }` the key will be "event" while I want it to be the value of the identifier "event".
For example, if `solidity` emits `compilationFinished` I want the object to be 
```typescript
{ compilationFinished: { /* content of compilation */ } }
```
You can try in your console : 
```typescript
const event = 'compilationFinished';
const memory = { [event]: 'something' };
console.log(memory); // { compilationFinished: 'something' }
```

**Pull request merged**